beacon-chain/rpc/prysm/v1alpha1/validator/proposer_deposits.go

```diff
@@ -277,3 +285,21 @@ func shouldRebuildTrie(totalDepCount, unFinalizedDeps uint64) bool {
 	unFinalizedCompute := unFinalizedDeps * params.BeaconConfig().DepositContractTreeDepth
 	return unFinalizedCompute > totalDepCount
 }
+
+// isLegacyDepositProcessPeriod determines if the current state should use the legacy deposit process.
+func isLegacyDepositProcessPeriod(beaconState state.BeaconState, canonicalEth1Data *ethpb.Eth1Data) bool {
+	// Before the Electra upgrade, always use the legacy deposit process.
+	if beaconState.Version() < version.Electra {
+		return true
+	}
+
+	// Handle the transition period between the legacy and the new deposit process.
+	requestsStartIndex, err := beaconState.DepositRequestsStartIndex()
+	if err != nil {
+		// If we can't get the deposit requests start index,
+		// we should default to the legacy deposit process.
+		return true
+	}
+	eth1DepositIndexLimit := math.Min(canonicalEth1Data.DepositCount, requestsStartIndex)
```

Comment:
not a blocker for PR but this means it's a check on the transition period between the legacy processing of deposits and new 6110 process